script/ghci-flags

```diff
@@ -3,63 +3,78 @@
-
-echo "-hide-all-packages"
-
-# Emit package flags from the environment file, removing comments & prefixing with -
-cabal exec --builddir=$repl_builddir -v0 bash -- -c 'cat $GHC_ENVIRONMENT' | grep -v '^--' | sed -e 's/^/-/'
-
-echo "-XHaskell2010"
-echo "-XStrictData"
-
-echo "-Wwarn"
-
-echo "-Weverything"
-echo "-Wno-all-missed-specialisations"
-echo "-Wno-implicit-prelude"
-echo "-Wno-missed-specialisations"
-echo "-Wno-missing-import-lists"
-echo "-Wno-missing-local-signatures"
-echo "-Wno-monomorphism-restriction"
-echo "-Wno-name-shadowing"
-echo "-Wno-safe"
-echo "-Wno-unsafe"
-echo "-Wno-star-is-type"
+# recent hie-bios requires us to output to the file at $HIE_BIOS_OUTPUT, but older builds & script/repl don’t set that var, so we default it to stdout
+output_file="${HIE_BIOS_OUTPUT:-/dev/stdout}"
+
+# do a build of dependencies up front to ensure they’re all available
+cabal v2-build -v0 all --only-dependencies
+
+build_products_dir="dist-newstyle/build/x86_64-osx/ghc-$ghc_version/build-repl"
+
+function flags {
+  # disable optimizations for faster loading
+  echo "-O0"
+  # don’t load .ghci files (for ghcide)
+  echo "-ignore-dot-ghci"
+
+  # where to put build products
+  echo "-outputdir $build_products_dir"
+  echo "-odir $build_products_dir"
+  echo "-hidir $build_products_dir"
+  echo "-stubdir $build_products_dir"
+
+  # preprocessor options, for -XCPP
+  echo "-optP-include"
+  echo "-optP$build_products_dir/autogen/cabal_macros.h"
+
+  # autogenerated sources, both .hs and .h (e.g. Foo_paths.hs)
+  echo "-i$build_products_dir/autogen"
+  echo "-I$build_products_dir/autogen"
+
+  # .hs source dirs
+  # TODO: would be nice to figure this out from cabal.project & the .cabal files
+  echo "-isemantic-analysis/src"
+  echo "-isemantic-ast/src"
+  echo "-isemantic-core/src"
+  echo "-isemantic-java/src"
+  echo "-isemantic-json/src"
+  echo "-isemantic-python/src"
+  echo "-isemantic-tags/src"
+  echo "-iapp"
+  echo "-isrc"
+  echo "-ibench"
+  echo "-itest"
+
+  # disable automatic selection of packages
+  echo "-hide-all-packages"
+
+  # run cabal and emit package flags from the environment file, removing comments & prefixing with -
+  cabal v2-exec -v0 bash -- -c 'cat "$GHC_ENVIRONMENT"' | grep -v '^--' | sed -e 's/^/-/'
+
+  # default language extensions
+  echo "-XHaskell2010"
+  echo "-XStrictData"
+
+  # treat warnings as warnings
+  echo "-Wwarn"
+
+  # default warning flags
+  echo "-Weverything"
+  echo "-Wno-all-missed-specialisations"
+  echo "-Wno-implicit-prelude"
+  echo "-Wno-missed-specialisations"
+  echo "-Wno-missing-import-lists"
+  echo "-Wno-missing-local-signatures"
+  echo "-Wno-monomorphism-restriction"
+  echo "-Wno-name-shadowing"
+  echo "-Wno-safe"
+  echo "-Wno-unsafe"
+  [[ "$ghc_version" = 8.6.* ]] || [[ "$ghc_version" = 8.8.* ]] && echo "-Wno-star-is-type" || true
+  [[ "$ghc_version" = 8.8.* ]] && echo "-Wno-missing-deriving-strategies" || true
```

Comment:
`semantic` only supports a single version of `ghc`, but this minimizes the amount of editing necessary when copying it to a new project.